jetbrains: Fix update.pl script for MPS and update

avdv opened the pull request:
###### Motivation for this change

New upstream releases.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

avdv committed:
jetbrains: Fix update.pl script for MPS

Jetbrains seems to have deviated from the usual URL pattern of
`/<product>/<version>/<name>-<version>.<ext>` by only using the major.minor part
of the version number in the first URL segment for MPS.

avdv committed:
jetbrains: Update

clion               : 2021.1.2 -> 2021.2
datagrip            : 2021.1.3 -> 2021.2
goland              : 2021.1.3 -> 2021.2
idea-community      : 2021.1.3 -> 2021.2
idea-ultimate       : 2021.1.3 -> 2021.2
mps                 : 2021.1   -> 2021.1.1
phpstorm            : 2021.1.4 -> 2021.2
pycharm-community   : 2021.1.3 -> 2021.2
pycharm-professional: 2021.1.3 -> 2021.2
rider               : 2021.1.3 -> 2021.1.5
ruby-mine           : 2021.1.2 -> 2021.2
webstorm            : 2021.1.2 -> 2021.2

r-rmcgibbo commented:
Result of `nixpkgs-review pr 132043` at 544e20e2 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>10 packages failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/f27704af57443c3dc347775445627a28">jetbrains.clion</a></li>
    <li><a href="https://gist.github.com/7bdef6a71978b5932170e7e07095abb6">jetbrains.datagrip</a></li>
    <li><a href="https://gist.github.com/f3ab64d4bfb7741f1d7b91ba8c924956">jetbrains.goland</a></li>
    <li><a href="https://gist.github.com/95ee54997a77dd10c06f85fdbe7aec85">jetbrains.mps</a></li>
    <li><a href="https://gist.github.com/cab101be95e5ab4f63b42efed9b86636">jetbrains.phpstorm</a></li>
    <li><a href="https://gist.github.com/fca1630f7d17a12b42d4ab7fe7279e7e">jetbrains.pycharm-community</a></li>
    <li><a href="https://gist.github.com/3e2ca76b38711b429f394ea387608dc2">jetbrains.pycharm-professional</a></li>
    <li><a href="https://gist.github.com/535b8973aa8e8cf64bae49a580b0808f">jetbrains.rider</a></li>
    <li><a href="https://gist.github.com/8e80135fcff1e56c0bd57bc5217c8a47">jetbrains.ruby-mine</a></li>
    <li><a href="https://gist.github.com/9aa70ed78f394d269d935b4c148a5b46">jetbrains.webstorm</a></li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement. Does something look off? Please file an [issue](https://github.com/rmcgibbo/r-rmcgibbo/issues) or reach out on IRC.

avdv commented:
AFAICS, these packages are x86_64-linux only. They also provide dmg packages for MacOS, but I don't know how we could deal with them.

Should we constrain the build to x86_64-linux here?

avdv commented:
Seems like they have changed the packaging, all the newer versions fail for some reason, such as:
```
patchelf: getting info about 'bin/fsnotifier64': No such file or directory
```
I'll look into this...

avdv committed:
jetbrains: Check for 64 bit specific fsnotifier

In April, 2021, Jetbrains announced the end of support for 32-bit OS
IntelliJ-based IDEs[1]:

> The final major version that will be guaranteed to run on a 32-bit OS will be
> v2021.1 for all IntelliJ-based IDEs, including AppCode, Clion, DataGrip, GoLand,
> IntelliJ IDEA, PhpStorm, PyCharm, Rider, RubyMine, and WebStorm.

[1]: https://blog.jetbrains.com/idea/2021/04/end-of-support-for-32-bit-operating-systems-in-intellij-based-ides/

SuperSandro2000 commented on pkgs/applications/editors/jetbrains/common.nix:
```diff
@@ -52,7 +52,7 @@ with stdenv; lib.makeOverridable mkDerivation rec {
 
       interpreter=$(echo ${stdenv.glibc.out}/lib/ld-linux*.so.2)
-      if [ "${stdenv.hostPlatform.system}" == "x86_64-linux" ]; then
+      if [ "${stdenv.hostPlatform.system}" == "x86_64-linux" -a -e bin/fsnotifier64 ]; then
```
```suggestion
      if [[ "${stdenv.hostPlatform.system}" == "x86_64-linux" && -e bin/fsnotifier64 ]]; then
```

avdv replied:
applied.

avdv commented:
> AFAICS, these packages are x86_64-linux only. They also provide dmg packages for MacOS, but I don't know how we could deal with them.
> 
> Should we constrain the build to x86_64-linux here?

Specifically, the build for jetbrains.idea-community.x86_64-darwin was last successful on 2020-03-14 08:14:10, which was for idea-community 2019.3.3 (https://hydra.nixos.org/build/148157165)

SebTM commented:
Tested with NixOS-Unstable in PHP/Bash-Environment - changes LGTM and working were working :)

fabianhjr commented:
Built and tested idea-community. LGTM

Ma27 commented:
@avdv would you mind opening a backport PR manually? :)

avdv commented:
> @avdv would you mind opening a backport PR manually? :)

Sure thing! Done: #133025
